cmd/ecr-credential-provider/plugin/plugin.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package main
+package plugin
```

Comment:
This change seems unrelated, can you revert it if it's not necessary for this PR?

Reply:
This solves a very peculiar thing I saw. Without it, compilation (aka `go build`) would fail if not performed inside `cmd/ecr-credential-provider` directory (requiring you to `cd`) and the error was that the `NewCredentialProvider` could not be found.

It all goes away when you refactor the `plugin.go` file out of the `main` package though. You can "script" around the issue but refactoring to a proper package seemed the better way to go. We can argue about the package location of course.